File: app/src/main/java/space/pal/sig/model/Fact.java
package space.pal.sig.model;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity(tableName = "facts")
public class Fact {

    @NonNull
    @PrimaryKey
    @ColumnInfo
    private String id;
    @NonNull
    @ColumnInfo
    private String name;

}

File: app/src/main/java/space/pal/sig/model/entity/Fact.kt
package space.pal.sig.model.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * SpaceNow
 * Created by Catalin on 12/24/2020
 **/
@Entity(tableName = "facts")
data class Fact(
        @ColumnInfo
        @PrimaryKey(autoGenerate = false)
        val content: String
)
